src/conf_mode/firewall.py

```diff
@@ -166,7 +185,8 @@ def get_config(config=None):
         # Update nat and policy-route as firewall groups were updated
         set_dependents('group_resync', conf)
 
-    if 'config_trap' in firewall and firewall['config_trap'] == 'enable':
+    #if 'config_trap' in firewall and firewall['config_trap'] == 'enable':
+    if 'config_trap' in firewall and firewall['global_options']['config_trap'] == 'enable':
```

Comment:
I think `config-trap` should  be migrated and removed as part of this refactor.

Always thought it strange that only the firewall node had an snmp trap for the config values, seems best implemented on a global level with the new config system.